src/Forms/ElementalAreaField.php

```diff
@@ -214,33 +217,85 @@ public function setSubmittedValue($value, $data = null)
         return $this->setValue(json_decode($value ?? '', true));
     }
 
+    /**
+     * This will perform FormField validation
+     * DataObject validation will happen in saveInto() as part of $element->write()
+     */
+    public function validate($validator)
+    {
+        $result = true;
+        foreach ($this->getElementData() as $arr) {
+            /** @var BaseElement $element */
+            /** @var array $data */
+            $element = $arr['element'];
+            $data = $arr['data'];
+            $elementForm = $this->getElementForm($element);
+            if (!$elementForm) {
+                continue;
+            }
+            $elementForm->loadDataFrom($data);
+            $formValidatorResult = $elementForm->validationResult();
+            if (!$formValidatorResult->isValid()) {
+                $validator->getResult()->combineAnd($formValidatorResult);
+                $result = false;
+            }
+        }
+        return $this->extendValidationResult($result, $validator);
+    }
+
     public function saveInto(DataObjectInterface $dataObject)
```

Comment:
What do the changes to the `saveInto()` block resolve? I tried changing this method back to what it was before this PR, and it seems to work exactly the same way as before.

Reply:
There's logic I moved to `private function getElementData()` which is shared with `validate()`